src/main/java/cubicchunks/asm/mixin/selectable/client/MixinRenderChunk_NoOptifine.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.asm.mixin.selectable.client;
+
+import static cubicchunks.client.RenderConstants.RENDER_CHUNK_SIZE;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.Constant;
+import org.spongepowered.asm.mixin.injection.ModifyConstant;
+
+import mcp.MethodsReturnNonnullByDefault;
+import net.minecraft.client.renderer.chunk.RenderChunk;
+/**
+ * Fixes renderEntities crashing when rendering cubes
+ * that are not at existing array index in chunk.getEntityLists(),
+ * <p>
+ * Allows to render cubes outside of 0..256 height range.
+ */
+@MethodsReturnNonnullByDefault
+@ParametersAreNonnullByDefault
+@Mixin(RenderChunk.class)
+public class MixinRenderChunk_NoOptifine {
+    
+    @Shadow private boolean needsUpdate;
+    
+    /**
+     * @author Foghrye4
+     * @reason Workaround to fix frame-freeze on block break
+     */
+    @Overwrite
+    public void setNeedsUpdate(boolean immediate) {
+        this.needsUpdate = true;
```

Comment:
This should be behind a **runtime-changable** configuration option to revert to vanilla behavior, with default being "true if using bigger chunks". Ideally this would be <whatever is set> | forge's "always rebuild chunks off-thread" option (whatever it's named). And you could submit forge PR to properly fix that option this way.